apps/storefront/src/pages/QuoteDraft/index.test.tsx

```diff
@@ -3854,4 +3855,263 @@ describe('when the user is a B2B customer', () => {
+        }),
+      },
+    });
+
+    const quoteInfo = buildQuoteInfoStateWith({
+      draftQuoteInfo: {
+        contactInfo: { email: customerEmail },
+      },
+      draftQuoteList: [product],
+    });
+
+    const getPreloadedState = (
+      billingAddress: Address = noAddress,
+      shippingAddress: Address = noAddress,
+    ) => ({
+      preloadedState: {
+        company: companyInfo,
+        storeInfo: storeInfoWithDateFormat,
+        quoteInfo: {
+          ...quoteInfo,
+          draftQuoteInfo: {
+            ...quoteInfo.draftQuoteInfo,
+            billingAddress,
+            shippingAddress,
+          },
+        },
+      },
+    });
+
+    const createQuoteMutation = vi.fn();
+
+    beforeEach(() => {
+      vi.clearAllMocks();
+      set(window, 'b2b.callbacks.dispatchEvent', vi.fn().mockReturnValue(true));
+      server.use(
+        graphql.query('Countries', () => HttpResponse.json({ data: { countries: [country] } })),
+        graphql.query('Addresses', () =>
+          HttpResponse.json({
+            data: {
+              addresses: {
+                totalCount: 2,
+                edges: [{ node: { ...billingAddress } }, { node: { ...shippingAddress } }],
+              },
+            },
+          }),
+        ),
+        graphql.query('getQuoteExtraFields', () =>
+          HttpResponse.json({ data: { quoteExtraFieldsConfig: [] } }),
+        ),
+        graphql.query('GetQuoteInfoB2B', () =>
+          HttpResponse.json({ data: { quote: { id: '272989', quoteNumber: '911911' } } }),
+        ),
+        http.post('*/api/v2/extra-fields/quote/validate', () => HttpResponse.json({ code: 200 })),
+        graphql.mutation('CreateQuote', async ({ request }: { request: Request }) => {
+          const body = await request.json();
+          createQuoteMutation(body.query);
+          return HttpResponse.json(
+            buildQuoteCreateResponseWith({
+              data: {
+                quoteCreate: { quote: { id: 123, createdAt: '1245' } },
+              },
+            }),
+          );
+        }),
+      );
+    });
+
+    it('should preserve billing address ID when selecting from saved addresses', async () => {
+      renderWithProviders(
+        <QuoteDraft setOpenPage={vi.fn()} />,
+        getPreloadedState(defaultBillingAddress, defaultShippingAddress),
+      );
+
+      await userEvent.click(screen.getByRole('button', { name: 'Edit info' }));
+
+      const billingFields = screen.getByRole('group', { name: 'Billing' });
+      await userEvent.click(within(billingFields).getByText('Choose from saved'));
+      await waitFor(() =>
+        expect(screen.getByRole('heading', { name: 'Choose from saved' })).toBeVisible(),
+      );
+      await userEvent.click(screen.getByRole('button', { name: 'Choose address' }));
+
+      await userEvent.click(screen.getByRole('button', { name: 'Save info' }));
+      await userEvent.click(await screen.findByRole('button', { name: 'Submit' }));
```

Comment:
I'm gonna imagine the submit button is available immediately, so no need to use `findBy`, `getBy` should suffice.